### PR TITLE
[Merged by Bors] - chore: Separate algebraic multiset lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Resplit `Algebra.BigOperators.Multiset.Basic` and `Algebra.BigOperators.Multiset.Lemmas` into two new files:
* `Algebra.BigOperators.Group.Multiset` for lemmas that require group-like structures (`Monoid`, `Group`, ...)
* `Algebra.BigOperators.Ring.Multiset` for lemmas that require ring-like structures (`MonoidWithZero`, `Ring`, ...)

Add `assert_not_exists Ring` in the former. Once #12974 lands, it will be strenghtenable to `assert_not_exists MonoidWithZero`.

---

- [x] depends on: #12836

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
According to a rough "auto-diff", no declarations were lost and the only added one is `natAbs_of_isUnit`.

File: Mathlib/Data/Finset/Card.lean [hunk 1]
```diff
@@ -26,9 +26,7 @@ This defines the cardinality of a `Finset` and provides induction principles for
 * `Finset.Nonempty.strong_induction`
 -/
 
--- TODO: After #12845,
--- assert_not_exists Ring
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Finset/Card.lean:30]: You can add this now

File: Mathlib/Data/Multiset/Bind.lean [hunk 2]
```diff
@@ -3,7 +3,7 @@ Copyright (c) 2017 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Multiset/Bind.lean:3]: Ring-free

**Review by Ruben-VandeVelde — APPROVED** [review]

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**